### ISSUE TITLE
Make it possible to append actions

### ISSUE THREAD

**Issue by johnpatcher** [fail2ban/fail2ban#989]
Maybe I've overlooked it in the documentation, but apparently there is no way to append an action to the already defined one. What I'm looking for is something like this:

A default banaction in the `[DEFAULT]` section:

    [DEFAULT]
    banaction = firewallcmd-ipset

This action is applied to any jail that does not define a custom `banaction`. However if I want additional banaction for particular jails (e.g. `sendmail-whois` for the `sshd`) jail, I need to set the `firewallcmd-ipset` action again:

    [sshd]
    enabled = true
    banaction = firewallcmd-ipset
                sendmail-whois[name=SSH]

I'm looking for something like an "append operator", e.g. "+=". It could look something like this:

    [sshd]
    enabled = true
    banaction += sendmail-whois[name=SSH]

Basically it would simply append the given action(s) to the already defined ones (i.e. the ones defined in the `[DEFAULT]` section.

Isn't this a quite common use case? Personally I hate the redundancy that is needed right now.

**Comment by yarikoptic** [comment]
I think it is resolved by recent (not yet released) new feature:
```
   - New interpolation feature for config readers - `%(known/parameter)s`.
     (means last known option with name `parameter`). This interpolation makes
     possible to extend a stock filter or jail regexp in .local file
     (opposite to simply set failregex/ignoreregex that overwrites it),
     see gh-867.
```

please reopen if I am wrong in my assessment ;)

**Comment by johnpatcher** [comment]
@yarikoptic As far as I can tell, #867 only adds the ability to add additional regular expressions, but not banactions. I'm looking for a similar facility to add additional banactions, e.g. something like `addbanaction`.

**Comment by yarikoptic** [comment]
Can't check atm but such feature should be in effect for any config file, so with this tippy could augment any file's setting in the corresponding .local file

**Comment by sebres** [comment]
Indeed, this new interpolation feature should work everywhere for the same variable name.
```
banaction = %(known/banaction)s 
            sendmail-whois[name=SSH]
```
But you can use "old" original interpolation also (with other variable name):
```
[DEFAULT]
my_banaction = firewallcmd-ipset
banaction = %(my_banaction)s

[sshd]
banaction = %(my_banaction)s 
            sendmail-whois[name=SSH]
```

**Comment by melikyuksel** [comment]
How does this work when you don't know the specific banaction (not specified in jail.local) or when it might change in the future? I want to add a secondary action, but the original banaction is dynamic. And I can't use the interpolation to access the old value because it's too recursive, as the error says.

**Comment by sebres** [comment]
> How does this work when you don't know the specific banaction (not specified in jail.local) 

Magic...
Parameter `banaction` (as well as other interpolation `action_`) are specified in `jail.conf` (and/or in distribution related includes), so it is "inherited" if the whole config becomes substituted.

Your entries in `jail.local` overwriting only the entries of `jail.conf` and other inherited includes.
It works like a merge process.

> And I can't use the interpolation to access the old value because it's too recursive, as the error says.

Wrong. 
You can do that... please take a closer look at my example - the magic word (prefix) is `known/`.

Following configs are valid (at least in v.0.10):
```ini
[jail1]
# only banaction (parameter only):
banaction = %(known/banaction)s[my_param="my_value"]

[jail2]
# extend action using action_ interpolation:
action = %(action_)s[my_param="my_value"]
         my_other_action[name=%(__name__)s, port="%(port)s", protocol="%(protocol)s", chain="%(chain)s"]

[jail3]
# extend action using last known action interpolation:
action = %(known/action)s[my_param="my_value"]
         my_other_action[name=%(__name__)s, port="%(port)s", protocol="%(protocol)s", chain="%(chain)s"]
```
